src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetHostName.cs

```diff
@@ -14,6 +14,11 @@ internal static partial class Sys
 
         internal static unsafe string GetHostName()
         {
+            if (OperatingSystem.IsWasi())
+            {
+                return "localhost";
+            }
```

Comment:
nit: it feels slightly wrong doing this special case in the Interop.GetHostName.cs managed binding layer, but calling into native code just for getting a constant string feels wrong too.

Btw. after looking at it some more I think we have an issue on Browser: we do use emscripten's gethostname() which returns the string `emscripten` and we just paper over it here: https://github.com/dotnet/runtime/blob/4cdbfdcca352e0f12e4829eb3fec638d717add51/src/libraries/System.Private.CoreLib/src/System/Environment.Browser.cs#L13-L15

But Interop.Sys.GetHostName() is used in more places and those would get the "wrong" hostname on Browser. I think I'd be in favor of unifying this and adding the special case for Browser here too.

Reply:
https://github.com/dotnet/runtime/issues/107741